### PR TITLE
Change alpine base image to 3.12 for a more stable version of MariaDB

### PR DESCRIPTION
**Pull request by fagguy**
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [✓] I have read the [contributing](https://github.com/linuxserver/docker-mariadb/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description:
Changed the base image to `linuxserver/baseimage-alpine:3.12`. And also updated related references `3.12`.

## Benefits of this PR and context:
This PR is a proposed fix to #76. Currently, there are a few applications, e.g. Nextcloud, which require a MariaDB version >= `10.2`.
Currently, the `latest` image based on Ubuntu Bionic gives us version `10.1`. The other available alternative is an image based on Alpine 3.13 (the `alpine` branch). However, as reported in #76, for armhf users, the container uses 100% CPU even when idle.

My proposal rebases `alpine` to `linuxserver/baseimage-alpine:3.12` which offers MariaDB `10.4`. And this setup does not result in the same problem as described above.

So Tl:dr, my proposal gives a working solution for armhf users, gives a newer version than the current `10.1`, and falls in line with the development direction to use Linux Alpine as a base image.

## How Has This Been Tested?
I have followed the [guidelines](https://github.com/linuxserver/docker-mariadb/blob/master/.github/CONTRIBUTING.md#testing-your-changes) and performed a build on my raspberry pi 4, running Raspian (32-bit).
I have tested that this image works with my Nextcloud container, without encountering 100% CPU usage.

## Source / References:
#76

### CODE REVIEW

**Comment by fagguy** [comment]
Well not sure why the build server fails. Here's my build logs.

```
pi@raspberrypi:~/docker-mariadb $ docker build   --no-cache   --pull   -t linuxserver/mariadb:latest .
Sending build context to Docker daemon  111.1kB
Step 1/12 : FROM ghcr.io/linuxserver/baseimage-alpine:3.12
3.12: Pulling from linuxserver/baseimage-alpine
Digest: sha256:8b840938805c0cbe4f5151b762ed4dcff5e046ef3ff682f8c4893694b25e58ef
Status: Image is up to date for ghcr.io/linuxserver/baseimage-alpine:3.12
 ---> c622daf37152
Step 2/12 : ARG BUILD_DATE
 ---> Running in 6acdccdc7b1d
Removing intermediate container 6acdccdc7b1d
 ---> a74de2b91a2c
Step 3/12 : ARG VERSION
 ---> Running in 08323b91058b
Removing intermediate container 08323b91058b
 ---> 3fd477a79252
Step 4/12 : ARG MARIADB_VERSION
 ---> Running in 86dcb9d71083
Removing intermediate container 86dcb9d71083
 ---> cf14398289d6
Step 5/12 : LABEL build_version="Linuxserver.io version:- ${VERSION} Build-date:- ${BUILD_DATE}"
 ---> Running in 08cfa0b632a8
Removing intermediate container 08cfa0b632a8
 ---> 1383380a70b4
Step 6/12 : LABEL maintainer="thelamer,nemchik"
 ---> Running in 7940398fb161
Removing intermediate container 7940398fb161
 ---> 73e3c139e48e
Step 7/12 : ENV MYSQL_DIR="/config"
 ---> Running in b0f1fd6723e2
Removing intermediate container b0f1fd6723e2
 ---> 20bc510903dd
Step 8/12 : ENV DATADIR=$MYSQL_DIR/databases
 ---> Running in f0f5a758669b
Removing intermediate container f0f5a758669b
 ---> 1cb9ddaca8f3
Step 9/12 : RUN  echo "**** install build packages ****" &&  apk add --no-cache --virtual=build-dependencies    curl &&  echo "**** install runtime packages ****" &&  if [ -z ${MARIADB_VERSION+x} ]; then                                                                   MARIADB_VERSION=$(curl -sL "http://dl-cdn.alpinelinux.org/alpine/v3.12/main/x86_64/APKINDEX.tar.gz" | tar -xz -C /tmp   && awk '/^P:mariadb$/,/V:/' /tmp/APKINDEX | sed -n 2p | sed 's/^V://');  fi &&  apk add --no-cache                                                     gnupg    mariadb==${MARIADB_VERSION}     mariadb-backup==${MARIADB_VERSION}      mariadb-client==${MARIADB_VERSION}     mariadb-common==${MARIADB_VERSION}                                                                                                                     mariadb-server-utils==${MARIADB_VERSION} &&  echo "**** cleanup ****" &&  apk del --purge        build-dependencies &&  rm -rf     /root/.cache     /tmp/* &&  mkdir -p                                                                                                        /var/lib/mysql
 ---> Running in 873a7cac0bf4
**** install build packages ****
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/armv7/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/armv7/APKINDEX.tar.gz
(1/4) Installing nghttp2-libs (1.41.0-r0)
(2/4) Installing libcurl (7.69.1-r3)
(3/4) Installing curl (7.69.1-r3)
(4/4) Installing build-dependencies (20210326.210027)
Executing busybox-1.31.1-r19.trigger
OK: 13 MiB in 34 packages
**** install runtime packages ****
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/armv7/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/armv7/APKINDEX.tar.gz
(1/35) Installing libgpg-error (1.37-r0)
(2/35) Installing libassuan (2.5.3-r0)
(3/35) Installing libcap (2.27-r0)
(4/35) Installing libgcc (9.3.0-r2)
(5/35) Installing libffi (3.3-r2)
(6/35) Installing libblkid (2.35.2-r0)
(7/35) Installing libmount (2.35.2-r0)
(8/35) Installing pcre (8.44-r0)
(9/35) Installing glib (2.64.6-r0)
(10/35) Installing libgcrypt (1.8.5-r0)
(11/35) Installing libsecret (0.20.3-r0)
(12/35) Installing pinentry (1.1.0-r2)
Executing pinentry-1.1.0-r2.post-install
(13/35) Installing libbz2 (1.0.8-r1)
(14/35) Installing gmp (6.2.0-r0)
(15/35) Installing nettle (3.5.1-r1)
(16/35) Installing p11-kit (0.23.22-r0)
(17/35) Installing libtasn1 (4.16.0-r1)
(18/35) Installing libunistring (0.9.10-r0)
(19/35) Installing gnutls (3.6.15-r1)
(20/35) Installing libksba (1.4.0-r0)
(21/35) Installing db (5.3.28-r1)
(22/35) Installing libsasl (2.1.27-r6)
(23/35) Installing libldap (2.4.50-r2)
(24/35) Installing npth (1.6-r0)
(25/35) Installing sqlite-libs (3.32.1-r0)
(26/35) Installing gnupg (2.2.23-r0)
(27/35) Installing mariadb-common (10.4.17-r1)
(28/35) Installing libaio (0.3.112-r1)
(29/35) Installing libstdc++ (9.3.0-r2)
(30/35) Installing libxml2 (2.9.10-r5)
(31/35) Installing mariadb (10.4.17-r1)
Executing mariadb-10.4.17-r1.pre-install
(32/35) Installing mariadb-backup (10.4.17-r1)
(33/35) Installing mariadb-client (10.4.17-r1)
(34/35) Installing perl (5.30.3-r0)
(35/35) Installing mariadb-server-utils (10.4.17-r1)
Executing busybox-1.31.1-r19.trigger
OK: 208 MiB in 69 packages
**** cleanup ****
WARNING: Ignoring APKINDEX.2c4ac24e.tar.gz: No such file or directory
WARNING: Ignoring APKINDEX.40a3604f.tar.gz: No such file or directory
(1/4) Purging build-dependencies (20210326.210027)
(2/4) Purging curl (7.69.1-r3)
(3/4) Purging libcurl (7.69.1-r3)
(4/4) Purging nghttp2-libs (1.41.0-r0)
Executing busybox-1.31.1-r19.trigger
OK: 207 MiB in 65 packages
Removing intermediate container 873a7cac0bf4
 ---> f896bc9b2293
Step 10/12 : COPY root/ /
 ---> c84d677d3bfb
Step 11/12 : EXPOSE 3306
 ---> Running in 925c3add1e6e
Removing intermediate container 925c3add1e6e
 ---> 6db16e91179f
Step 12/12 : VOLUME /config
 ---> Running in 105514f97845
Removing intermediate container 105514f97845
 ---> 975582720852
Successfully built 975582720852
Successfully tagged linuxserver/mariadb:latest
```

**Comment by fagguy** [comment]
How do I change the `--build-arg MARIADB_VERSION=10.5.8-r0` to `10.4.17-r1` on the build server?

**Comment by aptalca** [comment]
You can't. Jenkins ignores all changes to the jenkinsfile in PRs from outsiders as a security feature.

**Comment by fagguy** [comment]
Well... I'm currently running the image I built. It's functioning with my Nextcloud instance. And `docker stats` clearly shows that I have not run into #76.

```
version: "2.1"
services:
  swag:
    image: linuxserver/swag
    container_name: swag
    cap_add:
      - NET_ADMIN
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TIMEZONE}
      - URL=REDACTED
      - SUBDOMAINS=REDACTED
      - EXTRA_DOMAINS=REDACTED
      - ONLY_SUBDOMAINS=true
      - VALIDATION=http
    volumes:
      - /mnt/hdd/appdata/swag:/config
    ports:
      - 443:443
      - 80:80
    restart: unless-stopped

  nextcloud:
    image: linuxserver/nextcloud
    container_name: nextcloud
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TIMEZONE}
    volumes:
      - /mnt/hdd/appdata/nextcloud/config:/config
      - /mnt/hdd/appdata/nextcloud/data:/data
    depends_on:
      - mariadb
    restart: unless-stopped

  mariadb:
    image: linuxserver/mariadb:alpine-10.4.17-r1
    container_name: mariadb
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TIMEZONE}
      - MYSQL_ROOT_PASSWORD=${DB_ROOT_PW}
      - MYSQL_DATABASE=${DB_NAME}
      - MYSQL_USER=${DB_USER}
      - MYSQL_PASSWORD=${DB_PW}
    volumes:
      - /mnt/hdd/appdata/mariadb:/config
    restart: unless-stopped
```

**Comment by nemchik** [comment]
This lgtm, but there seems to be a newer package in alpine edge repos that resolves the issue. I am wondering if those will be released to 3.13 or if we'll have to wait for 3.14.

If it's likely the new versions of the packages will release to 3.13 we probably shouldn't need to merge this. If it's more likely that we'll have to wait for 3.14 then we'll merge this and hold at 3.12 and skip over 3.13 right to 3.14.

**Comment by fagguy** [comment]
Thanks for taking the time to review this.

I would prefer this to be merged in now, because there is no **official** solution from LSIO for armhf users at the moment. And @aptalca wasn't too pleased when I was "recommending random images that are not compatible". When the fix for Alpine `3.13` is confirmed and landed, you could just bump it back up to `3.13`.

Upgrading from MariaDB `10.4` -> `10.5` should be trivial (instructions below).
https://mariadb.com/kb/en/upgrading-from-mariadb-104-to-mariadb-105/

The only caveat is that this change might affect users who are already using MariaDB `10.5`. So I have a suggestion: Merge this in, but do not tag this as `mariadb:alpine`. Maybe tag it as `mariadb:alpine-version-10.4.17-r1` instead?

**Comment by nemchik** [comment]
Yeah the potential downgrade is what I'm trying to avoid. I don't know what effects that could have on user data. Releasing an additional tag wouldn't be too difficult, but we'd create a new branch for it to merge to and change a few things. I'll have a look at it maybe tomorrow and see about the changes.
